### PR TITLE
replace circleci2-builder image and fix tests

### PR DESCRIPTION
**Pull request by tobybellwood**
To reduce the number of image dependencies (and potential vulnerability surface), we can replace integratedexperts/circleci2-builder with docker/compose (and add in goss as a CI step.  If we end up creating our own CI image, we can switch to this.

This also removes the dependency on ahoy - switching to the docker-compose variants of the command

### CODE REVIEW

**Comment by simesy** [comment]
I'd like to extend this so that it can be run locally as well as remotely, as far as that makes sense.

**Comment by simesy** [comment]
Scratch my comments about locally versus remotely - not something I can fix.

**Comment by simesy** [comment]
Hacked my local a bit to run - this fails locally because (it seems) no git in docker/compose.

**Comment by tobybellwood** [comment]
force-pushed to remove rebase cruft - now a simple config.yml change.

**Comment by tobybellwood** [comment]
this will be replaced by a gitlab-ci.yml file in the next release.

**Comment by srowlands** [comment]
Can we change this PR base to `develop` and merge? Seems sane but notice this is still in draft + pointing at master

**Comment by tobybellwood** [comment]
TO-DO (follow up PR) - generate our own eicar file for maximum reliability

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,19 +22,20 @@ jobs:
       - run:
           name: Build project
           command: |
-            ahoy -v build
+            docker-compose up -d --build
```
Comment (steveworley) [inline .circleci/config.yml:25]: Should we keep these as `ahoy` as we're using ahoy everywhere else?

File: .circleci/config.yml [hunk 2]
```diff
@@ -47,17 +48,6 @@ jobs:
             else
               echo "Redis success: Redis contains $redisKeys keys."
             fi
-      - run:
-          name: Push Docker images to Dockerhub
-          command: |
-            if [ "${DEPLOY_ANY_BRANCH}" != "" ] || [ "${CIRCLE_BRANCH}" == "master" ] || [ ! -z ${CIRCLE_TAG} ]; then
```
Comment (steveworley) [inline .circleci/config.yml:53]: Good change, we should merge this in.

**Comment by AlexSkrypnyk** [comment]
This PR is almost a year old and a lot of changes were done since the. I think this PR should be abandoned:
- the CI image has already been replaced in https://github.com/govCMS/govcms8lagoon/pull/122
- the clamav tests are already in `develop`
- changes from `ahoy` to direct `docker-compose` are not justified - we have a unified way of running commands in all environments and in all CIs (CircleCI and GitLab) - it is not clear why we want to remove it.

@tobybellwood can you please review this PR and close it or comment on it. We are trying to clear up the queue.

**Comment by tobybellwood** [comment]
Happy to close, I'll submit the additional circleCI steps in another PR.

**Pull request closed without merging**